examples/src/test/kotlin/CodecTest.kt

```diff
@@ -0,0 +1,213 @@
+
+import com.mongodb.MongoClientSettings
+import com.mongodb.kotlin.client.coroutine.MongoClient
+import io.github.cdimascio.dotenv.dotenv
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
+import org.bson.BsonReader
+import org.bson.BsonType
+import org.bson.BsonWriter
+import org.bson.codecs.*
+import org.bson.codecs.configuration.CodecProvider
+import org.bson.codecs.configuration.CodecRegistries
+import org.bson.codecs.configuration.CodecRegistry
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import kotlin.test.assertEquals
+
+
+// :replace-start: {
+//    "terms": {
+//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string uri>\""
+//    }
+// }
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class CodecTest {
+    // :snippet-start: example-enum
+    enum class PowerStatus {
+        ON,
+        OFF
+    }
+    // :snippet-end:
+
+    // :snippet-start: example-codec
+    class PowerStatusCodec : Codec<PowerStatus> {
+        override fun encode(writer: BsonWriter, value: PowerStatus, encoderContext: EncoderContext) {
```

Comment:
These methods could also be shortened using equals syntax instead of a block syntax

Reply:
changed all single line fns to use equals syntax